### ISSUE TITLE
Provide event to access raw audio stream

### ISSUE THREAD

**Issue by nfiacco** [VapiAI/web#37]
Would be great to have an audio event like this: https://docs.retellai.com/make-calls/web-call#listen-to-events

**Comment by nikhilro** [comment]
It's a good idea but what's your use case?

**Comment by nfiacco** [comment]
To show a visualizer element! Honestly just a way of accessing the `Audio` element/ref would fantastic— because then we'd have a way to control the playback more granularly in addition to being able show a visualization.

**Comment by tomashanecak** [comment]
Agreed, that would be a great feature. I am currently struggling with visualizing user audio input. When I try to modify the node module I get ocasional weird loud static noise from the assistant. I suppose it might be some concurrent access to audio stream causing it in my case :/

**Comment by nikhilro** [comment]
i think we expose volume level updates if that helps

you can likely expose more stuff from daily webrtc connection directly. sdk is a thin wrapper on it. code here:  https://github.com/VapiAI/web/blob/main/vapi.ts

**Comment by OrPerl** [comment]
@nikhilro I will be happy to submit a PR, but I have never contributed before to an open source project.
I cloned the code locally, branched and committed changes locally, but cannot push remotely (as expected) to prepare the PR.
Any advise on the subject?

**Comment by OrPerl** [comment]
We had the same concern around how to build a proper waveform.
However vapi actually exposes a method called `getDailyCallObject`.
You can use that to get access to the daily object and from there listen to the `'track-started' event.
The event will expose the audio track that can be used to build the waveform.

**Comment by esme** [comment]
@nikhilro +1 on this issue. We're building a voice agent UI on top of the VAPI web SDK and need access to the underlying <audio> element to wire up volume/mute controls. Currently we're working around this by querying the DOM for audio[data-participant-id] after speech-start fires, which is fragile and could break if the SDK's DOM structure changes.

> However vapi actually exposes a method called getDailyCallObject.

getDailyCallObject() returns the Daily.js call object, but that doesn't give us a direct reference to the audio element either. An event (or a getter) that exposes the audio element or its stream would let us avoid DOM scraping entirely.
